src/jit/emitxarch.cpp

```diff
@@ -267,11 +267,20 @@ bool emitter::Is4ByteSSE4Instruction(instruction ins)
 bool emitter::TakesVexPrefix(instruction ins)
 {
     // special case vzeroupper as it requires 2-byte VEX prefix
-    // special case sfence and the prefetch instructions as they never take a VEX prefix
-    if ((ins == INS_vzeroupper) || (ins == INS_sfence) || (ins == INS_prefetcht0) || (ins == INS_prefetcht1) ||
-        (ins == INS_prefetcht2) || (ins == INS_prefetchnta))
+    // special case (l|m|s)fence and the prefetch instructions as they never take a VEX prefix
```

Comment:
nit: We could probably just say `special case the fencing and prefetch instructions...`